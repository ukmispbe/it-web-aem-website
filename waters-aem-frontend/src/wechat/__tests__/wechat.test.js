jest.mock('../../utils/modal/modal-portal');

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import WeChat, { weChatLinkClass } from '../index';

//Mocked Props
import DefaultProps from '../__mocks__/zh_CN/index';

describe('Feature: We Chat Component', () => {
    describe('Scenario Rendering We Chat in visible state', () => {

        describe('When rendering We Chat with the modal closed', () => {
            it('Then the snapshot should match', () => {
                const component = renderer.create(
                    <WeChat config={{ ...DefaultProps }} />
                );

                const root = component.root;
                const instance = root.instance;
                instance.setState({ isModalShown: true });
                expect(component).toMatchSnapshot();
            });
        });

        describe('When rendering We Chat with the modal open', () => {
            it('Then the snapshot should match', () => {
                const component = renderer.create(
                    <WeChat config={{ ...DefaultProps }} />
                );

                const root = component.root;
                const instance = root.instance;
                instance.setState({ isModalShown: false });
                expect(component).toMatchSnapshot();
            });
        });

        describe('When the We Chat social icon is clicked', () => {
            it('Then expect is to have been called and change modal state', () => {

                const component = shallow(<WeChat config={{ ...DefaultProps }} />);
                const weChatInstance = component.instance();
                const spyToggleModal = jest.spyOn(weChatInstance, 'toggleModal');
                const button = component.find(`.${weChatLinkClass}`)

                button.simulate('click', {
                    preventDefault: jest.fn(() => {})
                })

                expect(spyToggleModal).toHaveBeenCalled();
                expect(weChatInstance.state.isModalShown).toEqual(true);

            });
        });

    });
});
