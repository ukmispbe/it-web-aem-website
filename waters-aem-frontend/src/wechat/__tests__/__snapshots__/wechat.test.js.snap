// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: We Chat Component Scenario Rendering We Chat in visible state When rendering We Chat with the modal closed Then the snapshot should match 1`] = `
Array [
  <a
    className="cmp-footer-social-links__link"
    href="#"
    onClick={[Function]}
    target="_blank"
  >
    <div />
  </a>,
  <div
    className="cmp-modal-box "
  >
    <div
      className="cmp-modal"
    >
      <div
        className="cmp-modal__header header-with-title header-with-title-centered-tablet-and-over"
      >
        <div
          className="cmp-modal__title"
        >
          <div
            className="cmp-modal__title-text"
          >
            关注我们 – 沃特世中国
          </div>
        </div>
        <div
          className="cmp-modal__close-icon"
        >
          <div
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="cmp-wechat-modal"
      >
        <div
          className="cmp-wechat-modal__image"
        >
          <img
            alt="WeChat QR code"
            src="/content/dam/waters/zh/brand/product/wechat/WeChat.jpg"
          />
        </div>
        <div
          className="cmp-wechat-modal__text"
        >
          关注沃特世中国官方微信公众号，了解沃特世中国最新动态，获取最新资讯
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Feature: We Chat Component Scenario Rendering We Chat in visible state When rendering We Chat with the modal open Then the snapshot should match 1`] = `
<a
  className="cmp-footer-social-links__link"
  href="#"
  onClick={[Function]}
  target="_blank"
>
  <div />
</a>
`;
